doc/arch/adr-001-notify-integration.md

```diff
@@ -0,0 +1,424 @@
+  corresponding Email entity
+- Update SubscriptionContent entries to associate them with an email and
+  indicate they are processed
+- Liaise with EmailRenderer to convert one or more SubscriptionContent entities
+  into the subject and body for an Email
+
+#### Should
+
+- Take input of a frequency (immediate, weekly, daily)
+- Be able to consolidate SubscriptionContent entries that refer to the same
+  ContentChange
+
+#### Should not
+
+- Be able to process the same SubscriptionContent objects in multiple
+  concurrent processes
+
+#### May
+
+- Have its interface reconsidered to allow for increased ability to build
+  emails in parallel
+
+#### To determine
+- If a user has multiple subscriptions which are at different frequencies, is
+  this the class that will consolidate them?
+
+### DigestTimer
+
+**Responsibility**: Triggering EmailBuilderWorker to run at a given frequency
+
+#### Shall
+
+- Trigger EmailGenerationWorker to run at daily intervals
+- Trigger EmailGenerationWorker to run at weekly intervals
+
+#### To determine
+
+- When these intervals are
+- What mechanisms would be in place to handle this failing to occur
+
+### EmailRenderer
+
+**Responsibility**: Converting one or more Email Entries into a Notify Markdown
+formatted text
+
+#### Shall
+
+- Take input of one or more SubscriptionContent entities and a frequency
+- Generate a subject in plain text
+- Generate a body in Notify Markdown
+
+#### Should
+
+- Be able to generate differently formatted emails for weekly and daily emails
+
+#### May
+
+- Have sufficient information to include unsubscribe links in emails
+
+### EmailDeliveryWorker
+
+**Responsibility**: Finding Emails that should be sent and creating
+DeliveryAttempts for them and updating Email status accordingly
+
+#### Shall
+
+- Communicate with a transportation layer (Notify) to send an Email
+- Create a DeliveryAttempt object to associate with the Email
+- Update the Email and DeliveryAttempt objects based on the outcome of the
+  request
+
+#### Should
+
+- Be able to retry Emails that fail to send
+- Be able to abort trying to send an email if it’s a situation where retrying
+  won’t result in a success
+
+#### May
+
+- React to the transportation layer returning a 429 status code
+
+
+#### To determine
+
+- What should happen when an email has exhausted retries and failed
+
+### DeliveryMonitor
+
+**Responsibility**: Determining whether Notify was able to send emails we have
+tried to send or not
+
+#### Shall
+
+- Communicate with a transportation layer (Notify) to determine if a pending
+  delivery attempt is still in progress, successful or a failure
+- Update the Email and DeliveryAttempt objects based on the outcome of the
+  request
+
+#### Should
+
+- Be able to retry Emails that fail to send
+- Be able to abort trying to send an email if it’s a situation where retrying
+  won’t result in a success
+
+#### To determine
+
+- What should happen to stop a eternal retry loop?
```

Comment:
In which scenarios could this happen?

Reply:
Seems to just be (assuming our system is working correctly) if Notify returns notify-failure.